File: services/blog-service/src/main/java/com/portal/universe/blogservice/post/dto/PostResponse.java
package com.portal.universe.blogservice.post.dto;

import com.portal.universe.blogservice.post.domain.PostStatus;

import java.time.Instant;
import java.util.List;
import java.util.Set;

public record PostResponse(
        String id,
        String title,
        String content,
        String summary,
        String authorId,
        String authorUsername,
        String authorNickname,
        PostStatus status,
        Set<String> tags,
        String category,
        String metaDescription,
        String thumbnailUrl,
        List<String> images,
        Long viewCount,
        Long likeCount,
        Instant createdAt,
        Instant updatedAt,
        Instant publishedAt,

        // 기존 호환성 유지
        String productId
) {}

File: frontend/blog-frontend/src/dto/PostResponse.ts
// blog-frontend/src/dto/PostResponse.ts

export interface PostResponse {
  id: string;
  title: string;
  content: string;
  summary?: string;
  authorId: string;
  authorName?: string;
  status: string;
  tags?: string[];
  category?: string;
  viewCount?: number;
  likeCount?: number;
  publishedAt?: string;
  createdAt: string;
  updatedAt: string;
  metaDescription?: string;
  thumbnailUrl?: string;
}
